### PR TITLE
Fix vehicle doors after change model #578

### PR DESCRIPTION
**Pull request by FileEX**
This PR fixes issue #578

### CODE REVIEW

**Comment by botder** [comment]
Why were the previous PRs/commits reverted?

**Comment by FileEX** [comment]
> Why were the previous PRs/commits reverted?

https://github.com/multitheftauto/mtasa-blue/issues/1318 - It was 3 years ago, when I had very little experience in programming and instead of investigating why this is happening, I just removed the switch that was responsible for setting "missing doors" for vehicles i.e. tractor, caddy, etc. Now as you can see this the code is significantly different from that of the reverted PR.

File: Client/mods/deathmatch/logic/CClientVehicle.cpp [hunk 1]
```diff
@@ -1003,9 +1003,13 @@ void CClientVehicle::SetModelBlocking(unsigned short usModel, unsigned char ucVa
...
-        // Are we swapping from a vortex or skimmer?
-        bool bResetWheelAndDoorStates = (m_usModel == VT_VORTEX || m_usModel == VT_SKIMMER ||
-                                         (m_eVehicleType == CLIENTVEHICLE_PLANE && m_eVehicleType != CClientVehicleManager::GetVehicleType(usModel)));
+        // Are we swapping from a vehicle without doors?
+        bool bResetWheelAndDoorStates =
+            ((m_usModel == VT_VORTEX || m_usModel == VT_TRACTOR || m_usModel == VT_CADDY || m_usModel == VT_BAGGAGE || m_usModel == VT_BANDITO ||
+              m_usModel == VT_BFINJECT || m_usModel == VT_DOZER || m_usModel == VT_FORKLIFT || m_usModel == VT_KART || m_usModel == VT_MOWER ||
+              m_usModel == VT_QUAD || m_usModel == VT_RCBANDIT || m_usModel == VT_RCCAM || m_usModel == VT_RCGOBLIN || m_usModel == VT_RCRAIDER ||
+              m_usModel == VT_RCTIGER || m_usModel == VT_BLOODRA) ||
+             m_eVehicleType != CClientVehicleManager::GetVehicleType(usModel));
```
Comment (TheNormalnij) [inline Client/mods/deathmatch/logic/CClientVehicle.cpp]: Why don't you use `!HasDoors(m_usModel)` ?

Comment (FileEX) [reply Client/mods/deathmatch/logic/CClientVehicle.cpp]: It's a good idea

**Review by botder — APPROVED** [review]

**Pull request merged**